android/app/src/main/java/com/github/quarck/calnotify/monitorstorage/MonitorDatabase.kt

```diff
@@ -0,0 +1,205 @@
+//   Copyright (C) 2025 William Harris (wharris+cnplus@upscalews.com)
+//
+//   This program is free software; you can redistribute it and/or modify
+//   it under the terms of the GNU General Public License as published by
+//   the Free Software Foundation; either version 3 of the License, or
+//   (at your option) any later version.
+//
+//   This program is distributed in the hope that it will be useful,
+//   but WITHOUT ANY WARRANTY; without even the implied warranty of
+//   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//   GNU General Public License for more details.
+//
+//   You should have received a copy of the GNU General Public License
+//   along with this program; if not, write to the Free Software Foundation,
+//   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301  USA
+//
+
+package com.github.quarck.calnotify.monitorstorage
+
+import android.content.Context
+import android.database.sqlite.SQLiteDatabase
+import androidx.room.Database
+import androidx.room.Room
+import androidx.room.RoomDatabase
+import androidx.sqlite.db.SupportSQLiteDatabase
+import com.github.quarck.calnotify.database.CrSqliteRoomFactory
+import com.github.quarck.calnotify.logs.DevLog
+import java.io.File
+
+/**
+ * Room database for MonitorStorage.
+ * 
+ * Uses the existing "CalendarMonitor" database file to read existing data.
+ * Uses CrSqliteRoomFactory for cr-sqlite extension support.
+ * 
+ * MIGRATION NOTE: This Room database takes over from legacy SQLiteOpenHelper.
+ * Before Room opens, we migrate the legacy schema to add NOT NULL constraints
+ * on primary key columns (required by Room but missing in legacy schema).
+ */
+@Database(
+    entities = [MonitorAlertEntity::class],
+    version = 1,
+    exportSchema = false
+)
+abstract class MonitorDatabase : RoomDatabase() {
+    
+    abstract fun monitorAlertDao(): MonitorAlertDao
+
+    companion object {
+        private const val LOG_TAG = "MonitorDatabase"
+        private const val DATABASE_NAME = "CalendarMonitor"
+
+        @Volatile
+        private var INSTANCE: MonitorDatabase? = null
+
+        fun getInstance(context: Context): MonitorDatabase {
+            return INSTANCE ?: synchronized(this) {
+                INSTANCE ?: run {
+                    // Migrate legacy database BEFORE Room opens it
+                    migrateLegacyDatabaseIfNeeded(context)
+                    buildDatabase(context).also { INSTANCE = it }
+                }
+            }
+        }
+
+        private fun buildDatabase(context: Context): MonitorDatabase {
+            return Room.databaseBuilder(
+                context.applicationContext,
+                MonitorDatabase::class.java,
+                DATABASE_NAME
+            )
+                .openHelperFactory(CrSqliteRoomFactory())
+                .allowMainThreadQueries() // Match existing behavior - MonitorStorage uses sync queries
+                .addCallback(LoggingCallback())
```

Comment:
Do you still need this login call back? 

Also, we should have a good javadoc comment about the migration from the old database to the new one and the illegal state exception all that stuff